src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse_test.cc

```diff
@@ -0,0 +1,914 @@
+/*
+* Copyright 2018- The Pixie Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "src/stirling/utils/binary_decoder.h"
+#include "src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse.h"
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace mqtt {
+
+class MQTTParserTest : public ::testing::Test {
+};
+
+TEST_F(MQTTParserTest, Properties) {
+    Message frame;
+    ParseState result_state;
+    std::string_view frame_view;
+
+    uint8_t payload_format_indicator_publish[] = {
```

Comment:
Did these payloads come from real packet captures or are they handcrafted?

Reply:
They came from real packet captures. I used mosquitto to generate packets with different properties and payloads.